src/cliffordsynthesis/encoding/SATEncoder.cpp

```diff
@@ -114,7 +127,38 @@ Result SATEncoder::solve() const {
 void SATEncoder::extractResultsFromModel(Results& res) const {
   auto* const model = lb->getModel();
   tableauEncoder->extractTableauFromModel(res, T, *model);
-  gateEncoder->extractCircuitFromModel(res, *model);
+  auto qc = gateEncoder->extractCircuitFromModel(res, *model);
```

Comment:
This might create an unnecessary copy of the quantum circuit in case `ignoreRChanges=false`. It could be that RVO elides that copy, but I wouldn't be 100% sure.
Would it be cleaner to refactor the `Results` class so that it does not just store the OpenQASM string of the circuit and the string representation of the tableau, but the actual circuit and tableau? It would then produce the QASM and Tableau string only when requested.